src/MeshKernel/CurvilinearGridDeRefinement.cpp

```diff
@@ -44,18 +44,18 @@ meshkernel::CurvilinearGridDeRefinement::CurvilinearGridDeRefinement(std::shared
 meshkernel::CurvilinearGrid meshkernel::CurvilinearGridDeRefinement::Compute()
 {
     // Get the m and n indices from the point coordinates
-    auto [mFirstNode, nFirstNode] = m_grid->GetNodeIndices(m_firstPoint);
-    auto [mSecondNode, nSecondNode] = m_grid->GetNodeIndices(m_secondPoint);
+    auto const firstNode = m_grid->GetNodeIndices(m_firstPoint);
+    auto const secondNode = m_grid->GetNodeIndices(m_secondPoint);
 
     /// The points must lie on the same gridline
-    if (mSecondNode - mFirstNode != 0 && nSecondNode - nFirstNode != 0)
+    if (secondNode.m - firstNode.m != 0 && secondNode.n - firstNode.n != 0)
```

Comment:
now that you have a struct, can't  you make a function that checks this condition? that would be more expressive. 
say,
if (firstNode.IsSameGridline(SecondNode))